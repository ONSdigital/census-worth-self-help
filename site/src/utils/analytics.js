export default class Analytics {
  static trackEvent(category, action, name = "", value = "") {
    if (window._paq) {
      window._paq.push(["trackEvent", category, action, name, value])
    } else {
      console.debug(`Track event not set : ${category} - ${action} - ${name} - ${value}`)
    }
  }

  static trackSiteSearch(keyword, category, searchCount) {
    if (window._paq) {
      window._paq.push(["trackSiteSearch", keyword, category, searchCount])
    } else {
      console.debug(`Track site search : ${keyword} - ${category} - ${searchCount}`)
    }
  }

  static setPageType(pageType) {
    if (typeof window !== `undefined` && window._paq) {
      window._paq.push(["setCustomDimension", 1, pageType])
    }
  }
}