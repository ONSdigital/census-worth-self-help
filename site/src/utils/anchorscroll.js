export default (history, timeout = 1000) => {
  let observer
  let timeoutId

  const reset = () => {
    if (timeoutId) {
      clearTimeout(timeoutId)
      timeoutId = null
    }

    if (observer) {
      observer.disconnect()
    }
  }

  const convertTextToLink = text => {
    return text.toLowerCase().replace(/\s/g, "-")
  }

  const createScrollToElement = hash => {
    return () => {
      let headers = []
      let tags = ["h1", "h2", "h3"]

      tags.forEach(tag => {
        headers = headers.concat(Array.from(document.getElementsByTagName(tag)))
      })

      let element = headers.find(
        header => convertTextToLink(header.textContent) === hash
      )

      if (element) {
        element.scrollIntoView()
        reset()
        return true
      }

      return false
    }
  }

  function scroll(hash) {
    if (typeof hash !== "string") {
      return
    }

    const anchorText = hash.slice(1).toLowerCase()

    if (!anchorText) {
      return
    }

    const scrollToElement = createScrollToElement(anchorText)

    setTimeout(() => {
      if (scrollToElement()) {
        return
      }

      if (window.MutationObserver) {
        observer = new MutationObserver(scrollToElement)

        observer.observe(document, {
          attributes: true,
          childList: true,
          subtree: true
        })
      }

      timeoutId = setTimeout(reset, timeout)
    })
  }

  // listen for changes
  history.listen(({ location, action }) => {
    if (timeoutId) {
      reset()
    }

    if (action !== "POP") {
      return
    }

    if (location.hash) {
      scroll(location.hash)
    }
  })

  // also trigger on load
  if (history.location.hash) {
    scroll(history.location.hash)
  }
}
