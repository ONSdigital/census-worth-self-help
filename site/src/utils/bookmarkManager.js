export default class BookmarkManager {
  constructor() {
    this.bookmarks = JSON.parse(localStorage.getItem("bookmarks"))
    if (this.bookmarks === null) {
      this.bookmarks = []
    }
  }

  getTopBookmarks() {
    return this.bookmarks
  }

  isPageBookmarked(page) {
    return this.bookmarks.includes(page)
  }

  bookmarkPage(page) {
    if (!this.isPageBookmarked(page)) {
      this.bookmarks.push(page)
    }
    localStorage.setItem("bookmarks", JSON.stringify(this.bookmarks))
  }

  unBookmarkPage(page) {
    this.bookmarks = this.bookmarks.filter(function(bookmark) {
      return bookmark !== page
    })

    localStorage.setItem("bookmarks", JSON.stringify(this.bookmarks))
  }
}
