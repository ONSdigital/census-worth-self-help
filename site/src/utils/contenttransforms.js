var sanitizeHtml = require("sanitize-html")

export const transformSources = htmlString => {
  let videoPath = process.env.GATSBY_ASSETS_PATH
    ? process.env.GATSBY_ASSETS_PATH
    : ""
  if (!htmlString) {
    return ""
  }
  return htmlString.replace(/{{TARGET_ASSETS_SRC}}/g, videoPath)
}

export const htmlSanitize = htmlString => {
  return sanitizeHtml(htmlString, {
    allowedTags: sanitizeHtml.defaults.allowedTags.filter(function(tag) {
      return tag !== "iframe"
    }).concat([
      "video",
      "audio",
      "img",
      "source",
      "h1",
      "h2"
    ]),
    allowedAttributes: {
      a: ["href", "name", "target"],
      img: ["src", "title"],
      source: ["src", "type"],
      video: ["controls"],
      audio: ["controls"]
    }
  })
}
