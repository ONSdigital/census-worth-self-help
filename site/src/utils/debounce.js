import _ from "lodash"

// debounce with default delay, or return sync function if delay exactly zero (for unit tests)
export default (fn, delay) => {
  return (delay === 0)
    ? fn
    : _.debounce((...args) => fn(...args), delay || 1000)
}