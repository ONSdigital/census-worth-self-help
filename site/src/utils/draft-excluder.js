const READY_FOR_PRODUCTION = "Ready for Live Site"

// Articles in draft should not be deployed to production. An article is draft if the draftreason for the article
// is anything other than "Ready for production" - see ONS-359.
//
// We also use this Draft Excluder concept to not show articles that are for the call centre only.
//
// Initially this was done using the draft tag. This technique might be removed in the future.
exports.excludeDraftArticle = node => {
  return Boolean(
    process.env.EXCLUDE_DRAFTS && (
      (
        node.frontmatter.tags &&
        node.frontmatter.tags.includes("draft")
      ) || (
        node.frontmatter.draftreason &&
        node.frontmatter.draftreason !== READY_FOR_PRODUCTION
      ) || node.frontmatter.cconly
    )
  )
}
