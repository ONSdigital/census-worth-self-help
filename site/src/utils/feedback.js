export default class Feedback {
  static bookmarkClickEvent(article) {
    Feedback.feedbackEvent("click-event", "bookmark", article)
  }
  static articleWasBookmarked(article) {
    Feedback.feedbackEvent("article-was-bookmarked", "bookmarked", article, 1)
  }
  static articleWasUnBookmarked(article) {
    Feedback.feedbackEvent("article-was-bookmarked", "unbookmarked", article, 1)
  }
  static articleIsUseful(article, reason) {
    Feedback.feedbackEvent(
        "article-feedback-review",
        "POSITIVE REVIEW: " + article,
        reason
    )
    Feedback.feedbackEvent("article-feedback-rating", "rating", article, 1)
  }
  static articleIsNotUseful(article, reason) {
    Feedback.feedbackEvent(
      "article-feedback-review",
      "REVIEW: " + article,
      reason
    )
    Feedback.feedbackEvent("article-feedback-rating", "rating", article, -1)
  }
  static feedbackEvent(category, action, name = "", value = "") {
    if (window._paq) {
      window._paq.push(["trackEvent", category, action, name, value])
    }
  }
}
