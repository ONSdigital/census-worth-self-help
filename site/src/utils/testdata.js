export const data = {
  edges: [
    {
      node: {
        frontmatter: {
          title: "test Article 1",
          description: "testDescription",
          priority: 1,
          date: new Date()
        },
        fields: {
          pagename: "test-article-1",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 2",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular", "webchat"]
        },
        fields: {
          pagename: "test-article-2",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 3",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular"]
        },
        fields: {
          pagename: "test-article-3",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 4",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["draft"]
        },
        fields: {
          pagename: "test-article-4",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 5",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular"]
        },
        fields: {
          pagename: "test-article-5",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 6",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular"]
        },
        fields: {
          pagename: "test-article-6",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Directory 7",
          description: "testDescription",
          priority: 0,
          date: new Date()
        },
        fields: {
          pagename: "test-directory-1",
          collection: "directories"
        }
      }
    }
  ]
}

export const articleNode = data.edges[0].node
export const webchatArticleNode = data.edges[1].node
export const directoryNode = data.edges[2].node
export const draftNode = data.edges[3].node
export const popularList = {
  edges: [data.edges[1], data.edges[2], data.edges[4], data.edges[5]]
}
export const articleList = { edges: data.edges.slice(0, 6) }
