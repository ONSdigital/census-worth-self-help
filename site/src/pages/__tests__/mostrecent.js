import React from "react"
import renderer from "react-test-renderer"
import MostRecent from "../mostrecent"
import { render } from "react-testing-library"
import { articleList } from "../../utils/testdata"

describe("Most Recent", () => {
  it("renders correctly", () => {
    const mostRecentData = { allMarkdownRemark : articleList }
    const tree = renderer.create(<MostRecent data={mostRecentData}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})