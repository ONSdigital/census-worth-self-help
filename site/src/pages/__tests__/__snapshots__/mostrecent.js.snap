// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Most Recent renders correctly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "vkixka",
      "next": undefined,
      "styles": "
        height: 100vh;
        display: flex;
        flex-direction: column;
      ",
    }
  }
>
  <header
    css={
      Object {
        "map": undefined,
        "name": "181ra4m",
        "next": undefined,
        "styles": "
        width: 100vw;
        
    background: linear-gradient(to left, rgb(0,61,89), #33637a);
  ;
      ",
      }
    }
  >
    <div
      css={
        Object {
          "map": undefined,
          "name": "1thgah9",
          "next": undefined,
          "styles": "
            display: flex;
          ",
        }
      }
    >
      <div
        className="Header-Title-Style"
        css={
          Object {
            "map": undefined,
            "name": "17ih1bc",
            "next": undefined,
            "styles": "
              
    display: flex;
    justify-content: center;
    flex-direction: column;
  
              
    padding: 15px;
  
              flex-grow: 1;
            ",
          }
        }
        onClick={[Function]}
      >
        Self Help Facility
      </div>
      <div
        className="Header-button-Style"
        css={
          Object {
            "map": undefined,
            "name": "1sm5jo",
            "next": undefined,
            "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        background-color: rgb(255, 255, 255, 0.2);
        border-radius: 4px;
        margin: 10px 10px 10px 0px;
        min-width: 20px;
        object-fit: contain;
        text-align: center;
        height: 40px;
      ",
          }
        }
        data-testid="topbarlink-button"
        onClick={[Function]}
      >
        <div
          css={
            Object {
              "map": undefined,
              "name": "1dbjmka",
              "next": undefined,
              "styles": "
                  font-size: 19px;
                ",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-search fa-w-16 "
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </div>
      <div
        className="Header-button-Style"
        css={
          Object {
            "map": undefined,
            "name": "1sm5jo",
            "next": undefined,
            "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        background-color: rgb(255, 255, 255, 0.2);
        border-radius: 4px;
        margin: 10px 10px 10px 0px;
        min-width: 20px;
        object-fit: contain;
        text-align: center;
        height: 40px;
      ",
          }
        }
        data-testid="topbarlink-button"
        onClick={[Function]}
      >
        Menu
      </div>
    </div>
  </header>
  <div
    css={
      Object {
        "map": undefined,
        "name": "o9ztju",
        "next": undefined,
        "styles": "
          
    margin: 0 auto;
    max-width: 700px;
  
          overflow: scroll;
          width: 100vw;
          display: flex;
          flex-direction: column;
          flex-grow: 1;
        ",
      }
    }
  >
    <div
      css={
        Object {
          "map": undefined,
          "name": "1k0aln6",
          "next": undefined,
          "styles": "
            flex-grow: 1;
          ",
        }
      }
    >
      <div
        css={
          Object {
            "map": undefined,
            "name": "ypl6yy",
            "next": undefined,
            "styles": "
        border-bottom: 1px rgb(229, 229, 234) solid;
        padding: 15px 0px;
      ",
          }
        }
      >
        <div
          css={
            Object {
              "map": undefined,
              "name": "17k9f9t",
              "next": undefined,
              "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        border-radius: 4px;
        box-shadow: 0 2px 4px 0 rgba(212, 212, 212, 0.5);
        border: solid 1px #ececed;
        margin: 0px 15px;
        text-align: center;
        height: 50px;
        background-color: rgb(255, 255, 255);
        
      ",
            }
          }
          data-testid="large-button"
          onClick={[Function]}
        >
          <div
            css={
              Object {
                "map": undefined,
                "name": "1dj94zy",
                "next": undefined,
                "styles": "
            font-size: 19px;
            position: relative;
            left: 10px;
            color: rgb(0,61,89);
          ",
              }
            }
          >
            <div
              css={
                Object {
                  "map": undefined,
                  "name": "1wo7hj",
                  "next": undefined,
                  "styles": "
              position: absolute;
            ",
                }
              }
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-book fa-w-14 "
                data-icon="book"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
          <div
            className="Button-heading-Style"
            css={
              Object {
                "map": undefined,
                "name": "7tk8ru",
                "next": undefined,
                "styles": "
          margin: 0px 20px;
          color: rgb(0,61,89);
        ",
              }
            }
          >
            Explore Content
          </div>
        </div>
      </div>
      <div
        css={
          Object {
            "map": undefined,
            "name": "12ew49b",
            "next": undefined,
            "styles": "
        
    padding: 5px 15px;
  
      ",
          }
        }
      >
        <h1
          className="Article-Title-Style"
          css={
            Object {
              "map": undefined,
              "name": "1xcgh0k",
              "next": undefined,
              "styles": "
          margin-bottom: 0px;
          display: flex;
        ",
            }
          }
          data-testid="search-result-title"
        >
          <div
            css={
              Object {
                "map": undefined,
                "name": "8x4dpy",
                "next": undefined,
                "styles": "
              font-size: 19px;
              margin: 5px 20px 0px 2px;
            ",
              }
            }
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-clock fa-w-16 "
              data-icon="clock"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
          Recently Updated
        </h1>
      </div>
      <div
        className="Article-body-Style"
        css={
          Object {
            "map": undefined,
            "name": "bsbyq4",
            "next": undefined,
            "styles": "
        
    padding: 5px 15px;
  ;
        margin-bottom: 20px;
      ",
          }
        }
      >
        Most recent changes, need to check with Phil to see if there's actually content for this
      </div>
      <div
        css={
          Object {
            "map": undefined,
            "name": "1p7jpnp",
            "next": undefined,
            "styles": "
        
    padding: 15px 0px;
  ;
      ",
          }
        }
      >
        <div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "lcxyxv",
                "next": undefined,
                "styles": "
        
    padding: 15px;
    min-height: 30px;
  ;
        box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.5);
        border-left: 6px solid rgb(0,61,89);
        background-color: rgb(255, 255, 255);
      ",
              }
            }
            onClick={[Function]}
          >
            <div
              className="Card-heading-Style"
              css={
                Object {
                  "map": undefined,
                  "name": "qiy98z",
                  "next": undefined,
                  "styles": "
          color: inherit;
          flex-grow: 1;
          font-weight: semi-bold;
        ",
                }
              }
              data-testid="articletab-article-card"
            >
              test Article 1
            </div>
            <div
              className="Card-sub-head-Style-gray"
              css={
                Object {
                  "map": undefined,
                  "name": "1hejrb1",
                  "next": undefined,
                  "styles": "
          padding-top: 6px;
          display: flex;
          flex-direction: vertical;
          line-height: 1.08;
          max-height: 42.12px; // line-height * font-size * 3
        ",
                }
              }
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "lz6rtp",
                    "next": undefined,
                    "styles": "
            width: 70%;
            overflow: hidden;
            text-overflow: ellipsis;
            flex-grow: 1;
          ",
                  }
                }
              >
                testDescription
              </div>
              <div
                className="Card-meta-Style"
                css={
                  Object {
                    "map": undefined,
                    "name": "4j3tb8",
                    "next": undefined,
                    "styles": "
            display: flex;
            flex-direction: column;
            justify-content: flex-end;
          ",
                  }
                }
              >
                a few seconds ago
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "lcxyxv",
                "next": undefined,
                "styles": "
        
    padding: 15px;
    min-height: 30px;
  ;
        box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.5);
        border-left: 6px solid rgb(0,61,89);
        background-color: rgb(255, 255, 255);
      ",
              }
            }
            onClick={[Function]}
          >
            <div
              className="Card-heading-Style"
              css={
                Object {
                  "map": undefined,
                  "name": "qiy98z",
                  "next": undefined,
                  "styles": "
          color: inherit;
          flex-grow: 1;
          font-weight: semi-bold;
        ",
                }
              }
              data-testid="articletab-article-card"
            >
              test Article 2
            </div>
            <div
              className="Card-sub-head-Style-gray"
              css={
                Object {
                  "map": undefined,
                  "name": "1hejrb1",
                  "next": undefined,
                  "styles": "
          padding-top: 6px;
          display: flex;
          flex-direction: vertical;
          line-height: 1.08;
          max-height: 42.12px; // line-height * font-size * 3
        ",
                }
              }
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "lz6rtp",
                    "next": undefined,
                    "styles": "
            width: 70%;
            overflow: hidden;
            text-overflow: ellipsis;
            flex-grow: 1;
          ",
                  }
                }
              >
                testDescription
              </div>
              <div
                className="Card-meta-Style"
                css={
                  Object {
                    "map": undefined,
                    "name": "4j3tb8",
                    "next": undefined,
                    "styles": "
            display: flex;
            flex-direction: column;
            justify-content: flex-end;
          ",
                  }
                }
              >
                a few seconds ago
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "lcxyxv",
                "next": undefined,
                "styles": "
        
    padding: 15px;
    min-height: 30px;
  ;
        box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.5);
        border-left: 6px solid rgb(0,61,89);
        background-color: rgb(255, 255, 255);
      ",
              }
            }
            onClick={[Function]}
          >
            <div
              className="Card-heading-Style"
              css={
                Object {
                  "map": undefined,
                  "name": "qiy98z",
                  "next": undefined,
                  "styles": "
          color: inherit;
          flex-grow: 1;
          font-weight: semi-bold;
        ",
                }
              }
              data-testid="articletab-article-card"
            >
              test Article 3
            </div>
            <div
              className="Card-sub-head-Style-gray"
              css={
                Object {
                  "map": undefined,
                  "name": "1hejrb1",
                  "next": undefined,
                  "styles": "
          padding-top: 6px;
          display: flex;
          flex-direction: vertical;
          line-height: 1.08;
          max-height: 42.12px; // line-height * font-size * 3
        ",
                }
              }
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "lz6rtp",
                    "next": undefined,
                    "styles": "
            width: 70%;
            overflow: hidden;
            text-overflow: ellipsis;
            flex-grow: 1;
          ",
                  }
                }
              >
                testDescription
              </div>
              <div
                className="Card-meta-Style"
                css={
                  Object {
                    "map": undefined,
                    "name": "4j3tb8",
                    "next": undefined,
                    "styles": "
            display: flex;
            flex-direction: column;
            justify-content: flex-end;
          ",
                  }
                }
              >
                a few seconds ago
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "lcxyxv",
                "next": undefined,
                "styles": "
        
    padding: 15px;
    min-height: 30px;
  ;
        box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.5);
        border-left: 6px solid rgb(0,61,89);
        background-color: rgb(255, 255, 255);
      ",
              }
            }
            onClick={[Function]}
          >
            <div
              className="Card-heading-Style"
              css={
                Object {
                  "map": undefined,
                  "name": "qiy98z",
                  "next": undefined,
                  "styles": "
          color: inherit;
          flex-grow: 1;
          font-weight: semi-bold;
        ",
                }
              }
              data-testid="articletab-article-card"
            >
              test Article 4
            </div>
            <div
              className="Card-sub-head-Style-gray"
              css={
                Object {
                  "map": undefined,
                  "name": "1hejrb1",
                  "next": undefined,
                  "styles": "
          padding-top: 6px;
          display: flex;
          flex-direction: vertical;
          line-height: 1.08;
          max-height: 42.12px; // line-height * font-size * 3
        ",
                }
              }
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "lz6rtp",
                    "next": undefined,
                    "styles": "
            width: 70%;
            overflow: hidden;
            text-overflow: ellipsis;
            flex-grow: 1;
          ",
                  }
                }
              >
                testDescription
              </div>
              <div
                className="Card-meta-Style"
                css={
                  Object {
                    "map": undefined,
                    "name": "4j3tb8",
                    "next": undefined,
                    "styles": "
            display: flex;
            flex-direction: column;
            justify-content: flex-end;
          ",
                  }
                }
              >
                a few seconds ago
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        css={
          Object {
            "map": undefined,
            "name": "d1qmp1",
            "next": undefined,
            "styles": "
        padding: 15px 0px;
      ",
          }
        }
      >
        <div
          className="Button-subhead-Style"
          css={
            Object {
              "map": undefined,
              "name": "1dmo2ca",
              "next": undefined,
              "styles": "
          display: flex;
          
    padding: 5px 15px;
  
        ",
            }
          }
        >
          <div
            css={
              Object {
                "map": undefined,
                "name": "1k0aln6",
                "next": undefined,
                "styles": "
            flex-grow: 1;
          ",
              }
            }
            data-testid="pagination-element-showing"
          >
            Showing 
            1
            -
            4
             of 
            4
          </div>
          <div>
            Show 
            5
             per page
          </div>
        </div>
        <div
          css={
            Object {
              "map": undefined,
              "name": "1sf84gv",
              "next": undefined,
              "styles": "
          display: flex;
          justify-content: center;
        ",
            }
          }
        >
          <div
            className="Article-Title-Style"
            css={
              Object {
                "map": undefined,
                "name": "1j335gw",
                "next": undefined,
                "styles": "
        padding: 0px 5px;
        text-decoration: underline;
      ",
              }
            }
            data-testid="pagination-1-button"
            onClick={[Function]}
          >
            1
          </div>
        </div>
      </div>
    </div>
    <div
      css={
        Object {
          "map": undefined,
          "name": "ej75ms",
          "next": undefined,
          "styles": "
        
    padding: 30px 0px;
  
        width: 100%;
        padding: 10px 0px;
        background-color: rgb(236, 236, 237);
        display: flex;
        min-height: 100px;
        flex-direction: column;
        justify-content: center;
        flex-shrink: 0;
      ",
        }
      }
    >
      <img
        alt="Footer logo"
        css={
          Object {
            "map": undefined,
            "name": "1v6eegl",
            "next": undefined,
            "styles": "
          margin-bottom: 50px;
          margin-top: 50px;
        ",
          }
        }
        src="/footer_logo.svg"
      />
      <div
        className="Footer-Style"
      >
        © Office for National Statistics 2019
      </div>
    </div>
  </div>
</div>
`;
